include/fiction/layouts/bounding_box.hpp

```diff
@@ -47,95 +59,249 @@ class bounding_box_2d
+                    max = cell<Lyt>{std::max(max_cell.x, max_defect.x), std::max(max_cell.y, max_defect.y),
+                                    std::max(max_cell.z, max_defect.z)};
+                }
+            }
         }
         else
         {
-            // set min to max coordinate in the layout
-            min = {layout.x(), layout.y()};
+            min = coordinate<Lyt>{std::numeric_limits<decltype(coordinate<Lyt>::x)>::max(),
+                                  std::numeric_limits<decltype(coordinate<Lyt>::x)>::max()};
 
-            layout.foreach_coordinate(
-                [this](const auto& c)
-                {
-                    if (!is_empty_coordinate(c))
+            if constexpr (is_gate_level_layout_v<Lyt>)
+            {
+                layout.foreach_coordinate(
+                    [this](const auto& c)
                     {
-                        if (c.x < min.x)
+                        if (!is_empty_coordinate(c))
                         {
-                            min.x = c.x;
+                            if (c.x < min.x)
+                            {
+                                min.x = c.x;
+                            }
+                            if (c.y < min.y)
+                            {
+                                min.y = c.y;
+                            }
+                            if (c.x > max.x)
+                            {
+                                max.x = c.x;
+                            }
+                            if (c.y > max.y)
+                            {
+                                max.y = c.y;
+                            }
                         }
-                        if (c.y < min.y)
-                        {
-                            min.y = c.y;
-                        }
-                        if (c.x > max.x)
+                    });
+            }
+
+            if constexpr (is_cell_level_layout_v<Lyt>)
+            {
+                layout.foreach_cell(
+                    [this](const auto& c)
+                    {
+                        if (!layout.is_empty_cell(c))
                         {
-                            max.x = c.x;
+                            if (c.x < min.x)
+                            {
+                                min.x = c.x;
+                            }
+                            if (c.y < min.y)
+                            {
+                                min.y = c.y;
+                            }
+                            if (c.x > max.x)
+                            {
+                                max.x = c.x;
+                            }
+                            if (c.y > max.y)
+                            {
+                                max.y = c.y;
+                            }
                         }
-                        if (c.y > max.y)
+                    });
+            }
+
+            if constexpr (is_sidb_defect_surface_v<Lyt>)
```

Comment:
Maybe I'm missing something here because I must scroll up and down so much to cover the entirety of this function, but it looks like at first glance this entire condition and its content is copy-pasta from above. Is that so? Why?

Reply:
@marcelwa While the computations are similar, they are not identical because the processing differs between siqad and fiction coordinates. To simplify the implementation, we could restrict the input layouts to those based on fiction coordinates only.

Reply:
I'm not sure. It would definitely simplify the code here. Where would we get problems, though?

Reply:
@marcelwa I've tried to simplify it. I hope it is better now.